pygac/reader.py

```diff
@@ -733,12 +733,43 @@ def get_calibrated_dataset(self):
         return calibrated_ds
 
     def _georeference_data(self, calibrated_ds):
+        if not self.adjust_clock_drift:
+            thinned_lons, thinned_lats = self._get_lonlat_from_file()
+            gcps = np.array(((0, self.lonlat_sample_points[0]),
+                    (0, self.lonlat_sample_points[-1]),
+                    (len(self.scans) - 1, self.lonlat_sample_points[0]),
+                    (len(self.scans) - 1, self.lonlat_sample_points[-1])))
+            ref_lons = (thinned_lons[0, 0],
+                        thinned_lons[0, -1],
+                        thinned_lons[-1, 0],
+                        thinned_lons[-1, -1])
+            ref_lats = (thinned_lats[0, 0],
+                        thinned_lats[0, -1],
+                        thinned_lats[-1, 0],
+                        thinned_lats[-1, -1])
+            from pyorbital.geoloc_avhrr import estimate_time_offset
+            time_diff, _ = estimate_time_offset(gcps, ref_lons, ref_lats,
+                calibrated_ds["times"][0].values,
+                calibrated_ds.attrs["tle"],
+                calibrated_ds.attrs["max_scan_angle"])
+            time_diff = np.timedelta64(int(time_diff * 1e9), "ns")
```

Comment:
I think it would be nice to move the time offset estimation to a separate method

Reply:
Agreed

Reply:
fixed in [d76697f](https://github.com/pytroll/pygac/pull/151/commits/d76697f0fb9a2447b79d5c6bb85df51a00fa0ca7)